File: WebNovelApp/app/src/main/java/com/sinhvien/webnovelapp/models/BookmarkToggleRequest.kt
package com.sinhvien.webnovelapp.models

import com.google.gson.annotations.SerializedName

// Dữ liệu gửi đi khi nhấn nút Thêm/Xóa Bookmark
data class BookmarkToggleRequest(
    // NovelId là trường bắt buộc để biết truyện nào
    @SerializedName("NovelId") val novelId: Int,

    // Loại Bookmark (mặc định là Favorite)
    @SerializedName("BookmarkType") val bookmarkType: String? = "Favorite"

    // Lưu ý: KHÔNG cần ReaderId ở đây vì nó được lấy từ Token trên Server
)

File: DarkNovel/DarkNovel/Models/ApiModels/BookmarkToggleRequest.cs
namespace DarkNovel.Models.ApiModels
{
    // Phải khớp với Data Class 'BookmarkToggleRequest' bên Kotlin
    public class BookmarkToggleRequest
    {
        // Phải khớp với @SerializedName("NovelId")
        public int NovelId { get; set; }

        // Phải khớp với @SerializedName("BookmarkType")
        public string BookmarkType { get; set; }
    }
}